specs/CustomStoragePartition.md

```diff
@@ -0,0 +1,208 @@
+                        message << L"Failed: " << std::to_wstring(result) << L"(0x" << std::hex
+                                << result << L")" << std::endl;
+                        AsyncMessageBox(message.str(), L"Clear Custom Partition Data");
+                    }
+                    return S_OK;
+                }).Get()));
+          return S_OK;
+        }
+        
+```
+
+## .NET/WinRT
+```c#
+        private CoreWebView2 m_webview;
+        
+        // Sets custom storage partition identified by the partitionId, which uniquely
+        // identifies an application context.
+        void CoreWebView_Created()
+        {
+           // ...
+           // other WebView setup like add event handlers, update settings.
+           
+           // Sets custom storage partition identified by the partitionId, which uniquely
+           // identifies an application context.
+           string partitionId = "Partition 1";
+           m_webview.CustomStoragePartitionId = partitionId;
+           
+           // Navigate to start page
+           m_webview.Navigation(startPage);
+        }
+        
+        // Clears all data in custom storage partition identified by the partitionId.
+        // Called when the application context is removed.
+        async void ClearPartitionData(string partitionId)
+        {
+          await m_webview.Profile.ClearCustomStoragePartitionDataAsync(partitionId);
+          MessageBox.Show(this, "Completed", "Clear Custom Partition Data");
+        }
+
+```
+
+# API Details
+## Win32 C++
+```
+interface ICoreWebView2_18 : ICoreWebView2_17 {
+  /// Gets the `CustomStoragePartitionId` property.
+  [propget] HRESULT CustomStoragePartitionId([out, retval] LPWSTR* customStoragePartitionId);
+
+  /// Sets the `CustomStoragePartitionId` property.
+  /// This feature requires enabling 2 experimental browser features to work properly.
+  /// These features will be enabled by default in the future.
+  /// Before these features are enabled by default, please enable them by ensuring
+  /// `--enable-features=ThirdPartyStoragePartitioning,PartitionedCookies` is set in
+  /// `AdditionalBrowserArguments` in `CoreWebView2EnvironmentOptions` used to create
+  /// CoreWebView2Environment. If these features are not enabled, all data are treated
+  /// as unpartitioned and stored in the global default location for the profile.
+  /// When it is set, the page in the WebView will act as if the page were hosted in a
+  /// top level site uniquely associated with the `partitionId` and have a separate 
+  /// storage partition as described in https://developer.chrome.com/docs/privacy-sandbox/storage-partitioning/
+  /// and separete cookie partition as described in https://developer.chrome.com/docs/privacy-sandbox/chips/
+  /// with all cookies partitioned.
+  /// If `customStoragePartitionId` is nullptr or empty string, the
+  /// `CustomStoragePartitionId` will be reset and the page inside the WebView
```

Comment:
Does this occur specifically when the property is set to null/empty from a non-empty value?  Or can something trigger this any time while the property is empty?  What if I set alternating null and empty values?

Reply:
null and empty value are treated as the same. Set alternating null and empty values, does nothing, no changes.

Reply:
Please remove potentially confusing 'will be reset' part:
"... empty string, then the page inside the WebView ..."